lib/rules/length-zero-no-unit/index.js

```diff
@@ -40,9 +40,9 @@ function rule(actual, secondary, context) {
 			const ignorableIndexes = new Array(stringValue.length).fill(false);
 			const parsedValue = valueParser(stringValue);
 
-			parsedValue.walk((node, nodeIndex) => {
+			parsedValue.walk((node, nodeIndex, nodes) => {
 				if (decl.prop.toLowerCase() === 'font' && node.type === 'div' && node.value === '/') {
-					const lineHeightNode = parsedValue.nodes[nodeIndex + 1];
+					const lineHeightNode = nodes[nodeIndex + 1];
```

Comment:
When descending into functions, the `nodes` array will be scoped to the current function args. This is the array that `nodeIndex` refers to. [See the postcss-value-parser docs for details](https://github.com/TrySound/postcss-value-parser#valueparserwalknodes-callback-bubble). 

e.g., to parse the value `font: var(--foo, normal normal 400 16px/0 cursive);`, the value parser will run two loops. 

The outer loop operates on the following nodes (equivalent to `parsedValue.nodes`):

```
[
      { type: 'word', sourceIndex: 0, value: 'font' },
      { type: 'div', sourceIndex: 4, value: ':', before: '', after: ' ' },
      {
        type: 'function',
        sourceIndex: 6,
        value: 'var',
        before: '',
        after: '',
        nodes: [
          [Object], [Object],
          [Object], [Object],
          [Object], [Object],
          [Object], [Object],
          [Object], [Object],
          [Object], [Object],
          [Object]
        ]
      }
```

The inner loop (equivalent to `parsedValue.nodes[2].nodes`) runs on the following values:

```
[
      { type: 'word', sourceIndex: 10, value: '--foo' },
      { type: 'div', sourceIndex: 15, value: ',', before: '', after: ' ' },
      { type: 'word', sourceIndex: 17, value: 'normal' },
      { type: 'space', sourceIndex: 23, value: ' ' },
      { type: 'word', sourceIndex: 24, value: 'normal' },
      { type: 'space', sourceIndex: 30, value: ' ' },
      { type: 'word', sourceIndex: 31, value: '400' },
      { type: 'space', sourceIndex: 34, value: ' ' },
      { type: 'word', sourceIndex: 35, value: '16px' },
      { type: 'div', sourceIndex: 39, value: '/', before: '', after: '' },
      { type: 'word', sourceIndex: 40, value: '0' },
      { type: 'space', sourceIndex: 41, value: ' ' },
      { type: 'word', sourceIndex: 42, value: 'cursive' }
    ]
```

Using `nodes` here means we can pick the right node with `nodes[nodeIndex+1]`.